images/router/haproxy/conf/haproxy-config.template

```diff
@@ -463,13 +520,34 @@ backend be_secure_{{$cfgIdx}}
 
 {{/*--------------------------------- END OF HAPROXY CONFIG, BELOW ARE MAPPING FILES ------------------------*/}}
 {{/*
+    os_wildcard_domain.map: contains a mapping of wildcard hosts for a
+			[sub]domain regexps. This map is used to check if
+			a host matches a [sub]domain with has wildcard support.
+*/}}
+{{ define "/var/lib/haproxy/conf/os_wildcard_domain.map" }}
+{{     if matchPattern "true|TRUE" (env "ROUTER_ALLOW_WILDCARD_ROUTES" "")}}
+{{       range $idx, $cfg := .State }}
+{{         if ne $cfg.Host ""}}
+{{           if $cfg.IsWildcard }}
+{{genDomainWildcardRegexp $cfg.Host "" true}} 1
+{{           end }}
+{{         end }}
+{{       end }}
+{{     end }}{{/* end if router allows wildcard routes */}}
+{{ end }}{{/* end wildcard domain map template */}}
+
+{{/*
     os_http_be.map: contains a mapping of www.example.com -> <service name>.  This map is used to discover the correct backend
                         by attaching a prefix (be_http_) by use_backend statements if acls are matched.
 */}}
 {{ define "/var/lib/haproxy/conf/os_http_be.map" }}
 {{     range $idx, $cfg := .State }}
 {{       if and (ne $cfg.Host "") (eq $cfg.TLSTermination "")}}
+{{         if $cfg.IsWildcard }}
+{{genDomainWildcardRegexp $cfg.Host $cfg.Path false}} {{$idx}}
```

Comment:
looks like https://github.com/openshift/origin/pull/11201/files#diff-c7423e177dc76076aba057aa9cb302e3R132 prevents wildcard routes from being included at all if the router disables them